lib/services/messenger/translators/discourse.ts

```diff
@@ -259,7 +259,7 @@ export class DiscourseTranslator extends TranslatorScaffold implements Translato
 	private static convertReadConnectionResponse(
```

Comment:
Does `convertReadConnectionResponse()` ever do anything async? Has this not just pushed the same return issue for a Promise where one's not required up a level?

Also, I'm not very keen on having a multi-line map as a function parameter, I can understand stuff like `_.filter(things, (thing) => !thing`, but not a multi line map.

Reply:
Addressing the first point, this conforms to a more general pattern expected when translating a response, which sometimes is asynchronous (specifically translators/front.ts:366).  Issue created.

I've reached out in Flowdock about the second point.

Reply:
Cool, just wanted to check whether this could never be asynch, but if it can, it should be Promise based. Have spoken to you about the second item.

Reply:
Done.